data/types/base/pump/insulin_sensitivity.go

```diff
@@ -33,6 +33,12 @@ func (i *InsulinSensitivity) Validate(validator data.Validator) {
 }
 
 func (i *InsulinSensitivity) Normalize(normalizer data.Normalizer) {
+	if i.amountUnits == nil {
```

Comment:
For this type of thing were you are "secretly" passing an extra parameter down to Normalize (or Validate/Parse) there's no reason why you can't simply add it to the Normalize function, such as:

```
func (i *InsulinSensitivity) Normalize(normalizer data.Normalizer, units string)
```

After all, the only objects that MUST conform to the standard Normalize signature are the data types (data.Datum). Any sub-structs can use anything they want.